astronomer/providers/snowflake/operators/snowflake.py

```diff
@@ -73,9 +80,46 @@ class SnowflakeOperatorAsync(SnowflakeOperator):
     :param poll_interval: the interval in seconds to poll the query
     """  # noqa
 
-    def __init__(self, *, poll_interval: int = 5, **kwargs: Any) -> None:
+    def __init__(
+        self,
+        *,
+        snowflake_conn_id: str = "snowflake_default",
+        warehouse: Optional[str] = None,
+        database: Optional[str] = None,
+        role: Optional[str] = None,
+        schema: Optional[str] = None,
+        authenticator: Optional[str] = None,
+        session_parameters: Optional[Dict[str, Any]] = None,
+        poll_interval: int = 5,
+        **kwargs: Any,
+    ) -> None:
         self.poll_interval = poll_interval
-        super().__init__(**kwargs)
+        self.warehouse = warehouse
+        self.database = database
+        self.role = role
+        self.schema = schema
+        self.authenticator = authenticator
+        self.session_parameters = session_parameters
+        self.snowflake_conn_id = snowflake_conn_id
+        if self.__class__.__base__.__name__ != "SnowflakeOperator":
+            # It's better to do str check of the parent class name because currently SnowflakeOperator
+            # is deprecated and in future OSS SnowflakeOperator may be removed
+            if any(
```

Comment:
why this check?

Reply:
currently, SnowflakeOperator is deprecated and in the future OSS SnowflakeOperator may be removed to support backward compatibility I have added this